File: Contest_A/RoboX/RoboX.c
#include <stdio.h>
#include <string.h>

int main() {
    int t;
    scanf("%d", &t);

    while (t--) {
        int n, p;
        scanf("%d %d", &n, &p);

        char str[100];
        scanf("%s", str);

        int ans = (p % 2 == 0 ? 1 : 0);
        for (int i = 0; i < n; i++) {
            if (str[i] == 'R')
                p++;
            else if (str[i] == 'L')
                p--;

            if (p % 2 == 0)
                ans++;
        }

        printf("%d\n", ans);
    }

    return 0;
}

File: Contest_A/RoboX/RoboX.py
def main():
    t = int(input())

    while t > 0:
        n, p = map(int, input().split())

        str_input = input()
        ans = 1 if p % 2 == 0 else 0

        for char in str_input:
            if char == 'R':
                p += 1
            elif char == 'L':
                p -= 1

            if p % 2 == 0:
                ans += 1

        print(ans)

        t -= 1

if __name__ == "__main__":
    main()
